### PR TITLE
Auth: Remove Wagmi and use DynamicEmbeddedWidget

### PR DESCRIPTION
**Pull request by jmderby**
## Description

removing wagmi will reduce our bundle size by over half!

Screenshot 2024-11-19 at 3 44 33 PM
Screenshot 2024-11-19 at 4 37 45 PM

## Test plan

tested a built version via yalc and running quickstart example. 
tested a built version running dark mode, styles worked as expected

## Package updates

@crossmint/client-sdk-react-ui:minor

### CODE REVIEW

File: packages/client/auth/src/CrossmintAuthClient.ts [hunk 1]
```diff
@@ -232,7 +232,10 @@ export class CrossmintAuthClient extends CrossmintAuth {
 
     public async authenticateSmartWallet(address: string, signature: string) {
         try {
-            const queryParams = new URLSearchParams({ signinAuthenticationMethod: "evm" });
+            const queryParams = new URLSearchParams({
+                signinAuthenticationMethod: "evm",
+                callbackUrl: `${this.apiClient.baseUrl}/${AUTH_SDK_ROOT_ENDPOINT}/callback`,
```
Comment (jmderby) [inline packages/client/auth/src/CrossmintAuthClient.ts:237]: our backend still needs this callbackUrl (doesn't actually do anything anymore) but fixes an error we get client side.

File: packages/client/ui/react-ui/src/components/dynamic-xyz/DynamicContextProviderWrapper.tsx [hunk 2]
```diff
@@ -10,9 +10,12 @@ export interface DynamicContextProviderWrapperProps {
 
 export default function DynamicContextProviderWrapper({
     children,
-    settings,
+    settings = {},
```
Comment (mPaella) [inline packages/client/ui/react-ui/src/components/dynamic-xyz/DynamicContextProviderWrapper.tsx:13]: hm i dont get why these changes are needed? can u explain

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                },
+            }}
+            apiKeyEnvironment="development"
+        >
+            {children}
+        </DynamicContextProviderWrapper>
+    );
+}
+
+async function dynamicChainToCrossmintChain(
```
[the remark this answers is not in the document]
Comment (mPaella) [reply packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:73]: sure, lets just throw it to `/utils/dynamic/dynamicChainToCrossmintChain.ts`

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                        } catch (error) {
+                            console.error("[CryptoWalletConnectionHandler] Error authenticating wallet:", error);
+                            return false;
+                        }
+
+                        return true;
+                    },
+                },
+            }}
+            apiKeyEnvironment="development"
```
Comment (mPaella) [inline packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:66]: we need to be using the correct key env here, otherwise when they try to connect in prod it will fail saying "connect to a devnet network"

Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:66]: good catch!

File: packages/client/ui/react-ui/src/components/dynamic-xyz/DynamicContextProviderWrapper.tsx (hunk 2 above)
Comment (jmderby) [reply packages/client/ui/react-ui/src/components/dynamic-xyz/DynamicContextProviderWrapper.tsx:13]: I reverted these changes, I thought would be cleaner to set defaults but isn't

**Review by mPaella — APPROVED** [review]

**Comment by jmderby** [comment]
feel free to merge once glancing at my last commit changes

File: packages/client/auth/src/CrossmintAuthClient.ts (hunk 1 above)
Comment (AlbertoElias) [inline packages/client/auth/src/CrossmintAuthClient.ts:237]: we should remove it from the backend then and remove it here too. that was part of the point of the refactor and removing dead code. could you do quick PRs?

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                walletConnectors: [EthereumWalletConnectors, SolanaWalletConnectors],
+                events: {
+                    onAuthFlowCancel() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowCancel");
+                    },
+                    onAuthFlowClose() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowClose");
+                    },
+                    onAuthFailure(data, reason) {
+                        console.error("[CryptoWalletConnectionHandler] onAuthFailure", data, reason);
```
Comment (AlbertoElias) [inline packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:33]: shouldnt we use our new error handling here?

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                        console.log("[CryptoWalletConnectionHandler] onAuthSuccess", data);
+                    },
+                },
+                handlers: {
+                    handleConnectedWallet: async (wallet) => {
+                        console.log("[CryptoWalletConnectionHandler] handleConnectedWallet", wallet);
+
+                        const address = wallet.address;
+                        if (!address) {
+                            console.error("[CryptoWalletConnectionHandler] handleConnectedWallet: address is missing");
```
Comment (AlbertoElias) [inline packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:45]: and here

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                            return false;
+                        }
+
+                        const chain = await dynamicChainToCrossmintChain(wallet);
+                        console.log({ chain });
```
Comment (AlbertoElias) [inline packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:50]: do we need this log?

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx (hunk 5 above)
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:33]: Yes, i'll make a follow up ticket

File: packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx (hunk 7 above)
Comment (jmderby) [reply packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx:50]: this log was intentional

File: packages/client/auth/src/CrossmintAuthClient.ts (hunk 1 above)
Comment (jmderby) [reply packages/client/auth/src/CrossmintAuthClient.ts:237]: I've already got the backend PR in progress (just waiting on client migration)
